src/ZipkinBundle/Components/Messenger/ZipkinReceiveHandler.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+
+namespace ZipkinBundle\Components\Messenger;
+
+use Symfony\Component\Messenger\Envelope;
+use Zipkin\Propagation\Map;
+use Zipkin\Kind;
+use Zipkin\Tracing;
+
+class ZipkinReceiveHandler implements ZipkinHandlerInterface
+{
+    /**
+     * @var \Zipkin\Tracer
+     */
+    private $tracer;
+    /**
+     * @var callable
+     */
+    private $extractor;
+
+    public function __construct(Tracing $tracing)
+    {
+        $this->tracer = $tracing->getTracer();
+        $this->extractor = $tracing->getPropagation()->getExtractor(new Map());
+    }
+
+    public function handle(Envelope $envelope): Envelope
+    {
+        $stamp = $envelope->last(ZipkinStamp::class);
+        $context = $stamp? $stamp->getContext(): [];
+
+        $span = $this->tracer->nextSpan(($this->extractor)($context));
+        $span->start();
```

Comment:
I am curious about who is finishing this span. Also as a general question I'd like to ask is what kind of spans we do on messaging. I remember in kstreams app we trace the operations in the pipeline (map, filter, etc) so my question would be what are we tracing here. cc @jeqo

Reply:
I have to finish this part. In my Idea, I want to create a listener for [WorkerMessageHandledEvent](https://github.com/symfony/symfony/blob/5.0/src/Symfony/Component/Messenger/Event/WorkerMessageHandledEvent.php). WDYT

Reply:
for reference, all instrumentation in Java follow this pattern: 

https://github.com/openzipkin/brave/blob/b8ee69ef7a463a7ed364374536b2645529d508b9/instrumentation/jms/src/main/java/brave/jms/TracingConsumer.java#L46-L65

1. Extract and clean context from incoming request
2. Create a span to mark we have received a message `span.name("receive").kind(Span.Kind.CONSUMER);`
3. Inject context on the message again, so a processor/handler can extract and clear context to create a new span as part of your application. Here is an example: 

https://github.com/openzipkin/brave/blob/b8ee69ef7a463a7ed364374536b2645529d508b9/instrumentation/kafka-clients/src/main/java/brave/kafka/clients/KafkaTracing.java#L185-L202

and an example of how it looks like from the user side: https://github.com/openzipkin/brave/tree/master/instrumentation/kafka-clients#processing-messages